test/e2e-secret-tests.sh

```diff
@@ -24,27 +24,32 @@ source $(dirname "${BASH_SOURCE[0]}")/e2e-common.sh
 readonly E2E_TEST_NAMESPACE="default"
 
 # Constants used for creating ServiceAccount for the Control Plane if it's not running on Prow.
-readonly CONTROL_PLANE_SERVICE_ACCOUNT_NON_PROW_KEY_TEMP="$(mktemp)"
+readonly CONTROL_PLANE_GSA_NON_PROW_KEY_TEMP="$(mktemp)"
 
-# Constants used for creating ServiceAccount for Data Plane(Pub/Sub Admin) if it's not running on Prow.
-readonly PUBSUB_SERVICE_ACCOUNT_NON_PROW_KEY_TEMP="$(mktemp)"
+# Constants used for creating ServiceAccount for the Sources if it's not running on Prow.
+readonly SOURCES_GSA_NON_PROW_KEY_TEMP="$(mktemp)"
+
+# Constants used for creating ServiceAccount for the Broker if it's not running on Prow.
+readonly BROKER_GSA_NON_PROW_KEY_TEMP="$(mktemp)"
 
 # Constants used for authentication setup for GCP Broker if it's not running on Prow.
-readonly GCP_BROKER_SECRET_NAME="google-broker-key"
+readonly BROKER_GSA_SECRET_NAME="google-broker-key"
 
 function export_variable() {
   if (( ! IS_PROW )); then
-    readonly CONTROL_PLANE_SERVICE_ACCOUNT_KEY_TEMP="${CONTROL_PLANE_SERVICE_ACCOUNT_NON_PROW_KEY_TEMP}"
-    readonly PUBSUB_SERVICE_ACCOUNT_KEY_TEMP="${PUBSUB_SERVICE_ACCOUNT_NON_PROW_KEY_TEMP}"
+    readonly CONTROL_PLANE_GSA_KEY_TEMP="${CONTROL_PLANE_GSA_NON_PROW_KEY_TEMP}"
+    readonly SOURCES_GSA_KEY_TEMP="${SOURCES_GSA_NON_PROW_KEY_TEMP}"
+    readonly BROKER_GSA_KEY_TEMP="${BROKER_GSA_NON_PROW_KEY_TEMP}"
   else
-    readonly CONTROL_PLANE_SERVICE_ACCOUNT_KEY_TEMP="${GOOGLE_APPLICATION_CREDENTIALS}"
-    readonly PUBSUB_SERVICE_ACCOUNT_KEY_TEMP="${GOOGLE_APPLICATION_CREDENTIALS}"
+    readonly CONTROL_PLANE_GSA_KEY_TEMP="${GOOGLE_APPLICATION_CREDENTIALS}"
+    readonly SOURCES_GSA_KEY_TEMP="${GOOGLE_APPLICATION_CREDENTIALS}"
+    readonly BROKER_GSA_KEY_TEMP="${GOOGLE_APPLICATION_CREDENTIALS}"
   fi
 }
 
 # Setup resources common to all eventing tests.
 function test_setup() {
-  pubsub_setup "secret" || return 1
+  sources_auth_setup "secret" || return 1
```

Comment:
sth to think about, is "source" a better name than "pubsub"? What if we have a new source that does not use pubsub?

Reply:
> What if we have a new source that does not use pubsub?

I feel this is one more reason not to call the GSA variables or the service itself pubsub. This GSA doesn't only have the pubsub permission but also `cloudtrace.agent` & `monitoring.metricWriter`.  Also, `pubsub` doesn't tell anything about its user.

My guess if we have a new source that doesn't use pubsub is to still add the permission the sources GSA or create a new one.

Reply:
make sense.